packages/hooks/use-global-config/index.ts

```diff
@@ -53,11 +52,22 @@ export const provideGlobalConfig = (
   const context = computed(() => {
     const cfg = unref(config)
     if (!oldConfig?.value) return cfg
-    return merge(oldConfig.value, cfg)
+    return mergeConfig(oldConfig.value, cfg)
   })
   provideFn(configProviderContextKey, context)
   if (global || !globalConfig.value) {
     globalConfig.value = context.value
   }
   return context
 }
+
+const mergeConfig = <T extends Record<string, any>>(a: T, b: T) => {
+  const keys = [
+    ...new Set([...Object.keys(a), ...Object.keys(b)]),
+  ] as (keyof T)[]
+  const obj = {} as T
+  for (const key of keys) {
+    obj[key] = b[key] ?? a[key]
+  }
```

Comment:
Do you think that it is better that we use `_.merge` instead of writing this on our own?

Reply:
> _.merge({key:{a:'b',c:'d'}},{key:{a:'b'}})
```json
{
    "key": {
        "a": "b",
        "c": "d"
    }
}
```

the result of `mergeConfig` is
```
{
    "key": {
        "a": "b"
    }
}
```

Reply:
Also, `_.merge` will merge array, here is the example of offical
```js
var object = {
  'a': [{ 'b': 2 }, { 'd': 4 }]
};
 
var other = {
  'a': [{ 'c': 3 }, { 'e': 5 }]
};
 
_.merge(object, other);
// => { 'a': [{ 'b': 2, 'c': 3 }, { 'd': 4, 'e': 5 }] }
```

Reply:
Maybe we can try `webpack-merge` later, we will use this temporarily. Seems like you have some failure in testing.